sound/soc/sof/ipc.c

```diff
@@ -608,7 +608,8 @@ static void ipc_period_elapsed(struct snd_sof_dev *sdev, u32 msg_id)
 
 	if (spcm->pcm.compress)
 		snd_sof_compr_fragment_elapsed(stream->cstream);
-	else if (!stream->substream->runtime->no_period_wakeup)
+	else if (stream->substream->runtime &&
+		 !stream->substream->runtime->no_period_wakeup)
```

Comment:
The check seems OK but I wonder if we're missing something in the FW to cancel the position update after the stream is stopped. @lgirdwood any ideas?

Reply:
that hints at a race condition in the firmware, no?
Or that you didn't cancel the period workqueue?

Wondering why we never saw this with Intel firmware?

Reply:
I believe its a race condition. First thing is position update IPC message is put in a queue and sent to host at the idle task. So, there is no check whether stream is active or not while sending the position info in IPC. Even if we bump the priority to send the IPC message of position immediately, still there can be cases where the task switch is possible when Host component is executing, DSP is interrupted with IPC message for stop command. After updating the pipeline tasks status for stop the host component can resume and send the position update IPC command based on water mark level of buffer consumed before, stopped by scheduler to schedule next time.

Reply:
yeah there seems to be some issue within fw but as per my understanding we still need to add check for NULL pointer to avoid kernel panic or crash in any such cases.

Reply:
no disagreement on more protection at the kernel level, just wanted to make sure we don't lose track of the root cause.